src/audio/component.c

```diff
@@ -35,25 +35,17 @@ static const struct comp_driver *get_drv(uint32_t type)
 	struct list_item *clist;
 	const struct comp_driver *drv = NULL;
 	struct comp_driver_info *info;
-	uint32_t flags;
-
-	irq_local_disable(flags);
```

Comment:
what happens if new drivers are being added to this list via topology ?

Reply:
No, this function is to get driver info only, if we want to add driver at runtime, we need to implement new function for that, e.g.

```
add_drv()
{
    irq_local_disable();
    if (!get_drv())
       add_drv_to_list();
    irq_local_enable();
}
```

Reply:
`platform_shared_commit()` must be called in any case to invalidate cache and make sure that fresh copy of data is fetched on the next access.
And atm there is no such thing like RO access to SHARED_DATA, `platform_shared_commit()` would both invalidate and writeback on a platform with no uncache access.
A global lock (instead of a global critical section) would be required anyway to make sure the list is not corrupted if there is insert/remove operation performed at the same time (not present yet, but likely to happen when loadable modules are introduced).

Reply:
> `platform_shared_commit()` must be called in any case to invalidate cache and make sure that fresh copy of data is fetched on the next access.
> And atm there is no such thing like RO access to SHARED_DATA, `platform_shared_commit()` would both invalidate and writeback on a platform with no uncache access.
> A global lock (instead of a global critical section) would be required anyway to make sure the list is not corrupted if there is insert/remove operation performed at the same time (not present yet, but likely to happen when loadable modules are introduced).

Thanks for comment, @mmaka1 let me close this PR as it is not a mandatory change for the UUID update.